Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+
+/// <summary>
+/// The class that handles which UI is open or not at a time
+/// </summary>
+public class UIManager : MonoBehaviour
+{
+    public static UIManager Instance;
+
+    private bool isGUIOpen = false;
+    public bool IsGUIOpen { get => isGUIOpen; }
+
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+            return;
+        }
+
+        Instance = this;
+    }
+
+    private void Start()
+    {
+        InputSystem.actions.FindAction("ToggleInventory").performed += ctx => InventoryKeyPress();
+        InputSystem.actions.FindAction("ToggleStore").performed += ctx => StoreKeyPress();
+
+        InventoryManager.Instance.SetInventoryActiveState(false);
+        StoreManager.Instance.SetStoreActiveState(false);
+    }
+
+    // Handles the inventory keybind press(call this from anywhere you want to toggle the inventory)
+    private void InventoryKeyPress()
+    {
+       SetInventoryState(!InventoryManager.Instance.InventoryGUI.activeSelf);
+    }
+
+    // Handles the store keybind press (call this from anywhere you want to toggle the store)
+    private void StoreKeyPress()
+    {
+        SetStoreState(!StoreManager.Instance.StoreGUI.activeSelf);
+    }
+
+    // Method that is called to open/close the inventory(call this from anywhere you want to set the inventory)
+    public void SetInventoryState(bool state)
```

Comment:
This works for now, but I feel as if this could be simplified further to prevent duplicating code for every GUI instance. For example, instead of a creating a method per instance you could provide a parameter for the intended instance.